libraries/dictionary/README.md

```diff
@@ -1,9 +1,261 @@
+
+The restrictions property of a field can have a value that is either a single restrictions object, or an array with any number of restrictions objects. If an array of restriction objects is provided, each set of restrictions will be applied in turn - for data to be valid, all restrictions in the array must pass. A restrictions object can either contain a set of restrictions from the table below, or be a [conditional restriction](#conditional-restrictions).
+
+The full list of available restrictions are:
+
+| Restriction | Used with Field Types         | Type                                                            | Description                                                                                                                                                                                                                                                                                                                           | Examples                                                                                                                                                          |
+| ----------- | ----------------------------- | --------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `codeList`  | `integer`, `number`, `string` | Array of type of the field                                      | An array of values of the type matching this field. Data provided for this field must have one of the values in this list.                                                                                                                                                                                                            | `["Weak", "Average", "Strong"]`                                                                                                                                   |
+| `compare`   | all                           | [ComparedFieldsRule](#comparedfieldsrule-data-structure) object | Enforces that this field has a value based on the provided value in another field. Examples would be to ensure that the two values are not equal, or for numeric values ensure one is greater than the other.                                                                                                                         | `{ "fields": ["age_at_diagnosis"], "relation": "greaterThanOrEqual" }` Ensure that a field such as `age_at_death` is greater than the provided `age_at_diagnosis` |
```

Comment:
Example uses `greaterThanOrEqual`. 

But don't see this value in:
https://github.com/overture-stack/lectern/blob/da452d4dc016daa15738988d804547e47dfcc872/libraries/dictionary/README.md#comparedfieldsrule-data-structure

Also snakecase or camelcase?

Reply:
we want to consistently use camelCase for json properties.

I might've missed this property in the docs, thanks for pointing it out.